api/pkg/backend/service.go

```diff
@@ -447,3 +460,342 @@ func (s *APIService) listStorageType(ctx context.Context, request *restful.Reque
 	storageTypes, err := s.backendClient.ListType(ctx, listTypeRequest)
 	return storageTypes, err
 }
+
+//tiering functions
+func (s *APIService) CreateTier(request *restful.Request, response *restful.Response) {
+	log.Info("Received request for creating tier.")
+	tier := &backend.Tier{}
+	body := ReadBody(request)
+	err := json.Unmarshal(body, &tier)
+	if err != nil {
+		log.Error("error occurred while decoding body", err)
+		response.WriteError(http.StatusBadRequest, nil)
+		return
+	}
+	if len(tier.Name) == 0 {
+		errMsg := fmt.Sprintf("\"Name\" is required parameter should not be empty")
+		log.Error(errMsg)
+		response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+		return
+	}
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	actx := request.Attribute(c.KContext).(*c.Context)
+	tier.TenantId = actx.TenantId
+
+	//list tiers
+	tierResult, err := s.backendClient.ListTiers(ctx, &backend.ListTierRequest{})
+	log.Info("The list of existing tiers:", tierResult)
+	if err != nil {
+		log.Errorf("failed to list Tiers: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	//validate name
+	for _, t := range tierResult.Tiers {
+		if tier.Name == t.GetName() {
+			errMsg := fmt.Sprintf("name: %v is already exists.", tier.Name)
+			log.Error(errMsg)
+			response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+			return
+		}
+	}
+
+	//duplicate backends check:
+	visited := make(map[string]int, 0)
+	for _, val := range tier.Backends {
+
+		visited[val]++
+	}
+
+	var duplicates []string
+	for u, v := range visited {
```

Comment:
can these be written in the for loop at 508? simply if visited[val] > 0: duplicates = append(duplicates, val)

Reply:
done